File: src/main/java/org/distril/beengine/command/data/Args.java
package org.distril.beengine.command.data;

import lombok.AllArgsConstructor;
import org.distril.beengine.player.Player;
import org.distril.beengine.player.data.GameMode;
import org.distril.beengine.server.Server;

import java.util.Map;
import java.util.Random;

@AllArgsConstructor
public class Args {

	private final Map<String, String> args;

	public Integer getInteger(String key) {
		try {
			return Integer.parseInt(this.args.get(key));
		} catch (NumberFormatException exception) {
			return null;
		}
	}

	public Float getFloat(String key) {
		try {
			return Float.parseFloat(this.args.get(key));
		} catch (NumberFormatException exception) {
			return null;
		}
	}

	public Player getTarget(String key) {
		var username = this.getString(key);

		var server = Server.getInstance();
		if (username != null) {
			var players = server.getPlayers();
			if (username.equals("@r")) {
				return players.stream().toList().get(new Random().nextInt(players.size()));
			}
		}

		return server.getPlayer(username);
	}

	public String getString(String key) {
		return this.args.get(key);
	}

	public GameMode getGameMode(String key) {
		var id = this.getInteger(key);
		if (id != null) {
			return GameMode.fromId(id);
		}

		return GameMode.fromIdentifierOrAlias(this.getString(key));
	}

	public boolean has(String key) {
		return this.args.containsKey(key);
	}

	public boolean isEmpty() {
		return this.args.isEmpty();
	}
}

File: src/main/java/org/distril/beengine/command/data/Args.kt
package org.distril.beengine.command.data

import org.distril.beengine.player.Player
import org.distril.beengine.player.data.GameMode
import org.distril.beengine.server.Server
import java.util.*

class Args(private val args: Map<String, String>) {

	fun getInteger(key: String): Int? = try {
		this.args[key]!!.toInt()
	} catch (exception: NumberFormatException) {
		null
	}

	fun getFloat(key: String): Float? = try {
		this.args[key]!!.toFloat()
	} catch (exception: NumberFormatException) {
		null
	}

	fun getTarget(key: String): Player? {
		val username = this.getString(key)!!

		val players = Server.players
		if (username == "@r") {
			return players.stream().toList()[Random().nextInt(players.size)]
		}

		return Server.getPlayer(username)
	}

	fun getString(key: String) = this.args[key]

	fun getGameMode(key: String): GameMode? {
		val id = this.getInteger(key)
		return if (id != null) GameMode.fromId(id) else GameMode.fromIdentifierOrAlias(this.getString(key)!!)
	}

	fun has(key: String) = this.args.containsKey(key)

	fun isEmpty() = this.args.isEmpty()
}
